### PR TITLE
Resolve issue #5 and and updated board properties

### PR DESCRIPTION
**Pull request by haydenclev**
This time with spaces.

### CODE REVIEW

**Comment by haydenclev** [comment]
Is there a reason this pull request isn't being accepted? I would like to work on it and make a fix if so. The get_all_boards function is critical to something I am trying to implement using this library.

**Comment by antonizoon** [comment]
Nah, we're just kind of slow. I was having a team member review it but I guess better to do it myself when I have time, which could be a while. At a glance it is very good and some things are applicable to BASC-py4chan as well. I guess if you do have the time could you summarize all your changes so I can figure out what it's all about?

**Comment by haydenclev** [comment]
Oh good to hear! I was worried it was dismissed for some reason and wanted to know how it could be fixed. Changes are summarized below:

1. Lots of basic syntax updates (eg turning "4chan" into "8chan")
2. Updated the get_all_boards() function to properly handle the 8chan api json response and return a list of Board objects
3. Included an example usage of get_all_boards() for the docs
4. Added all the metadata contained about each board to the Board object using the @property decorator. Fairly constant data like title and tags are cached while changeable things like number of active users and time make requests to 8chan again using  _get_metadata()

Hope that clears things up! Let me know if there's anything else that can be tweaked/improved.

**Comment by antonizoon** [comment]
Alright, basically you fixed py8chan to work with their new catalog system. Well that is awesome so I will just merge and publish it. 

And hey if you want to try and do some more work on this such as building a [doctest system](https://docs.python.org/2/library/doctest.html) (a very elegant Python method for test driven development) or other interesting things, building these sets of API wrappers taught me everything I had to know about Python, not my masters degree. 

Primarily our team is working to build an alternative to the aging Java-based Asagi scraper at https://github.com/bibanon/eve . I think that if we integrate these API wrappers with that it would make it possible to have a near universal chan or even forum scraper system as long as it uses the same class methods. Maybe that might be good to work together with on the future?

https://docs.python.org/2/library/doctest.html

**Comment by antonizoon** [comment]
Alright I published a new version on pypi, v0.0.4, thank you for your contribution.

One regression I was having after merging woodenphone's pull request was that this unicode assertion wasn't working. What's up with it? I disabled it for this release.

https://github.com/bibanon/py8chan/pull/13

https://github.com/bibanon/py8chan/commit/6fb0fa836963f2bf5d8b9ecd35b45eab5ad2e3b3

BTW there were these two intriguing code suggestions that were never packed into a pull request so I never got around to them. Could you experiment with them and see if they are good to merge?

https://github.com/bibanon/py8chan/issues/11

https://github.com/bibanon/py8chan/issues/12

**Comment by haydenclev** [comment]
Sorry for a late reply. Stoked to see the pull request accepted! I would
definitely like to work on more features in the future. The Eve project
sounds really interesting as well. Do you see it performing the work of the
py8chan API wrapper in the future? If that's the case it might make more
sense to focus my efforts there and contribute to a more universal tool.

In regards to the specifics you pointed out:

#13 : My best guess is the assertion is a typo. Instead of testing 'board'
it should test 'uri'. It doesn't make much sense once you realize a few
lines up there is another assertion test for 'board' against a different
outcome so one of the two is bound to fail.

#11 : The fix for poster_id is solid so I implemented it as well. I think I
ran into a new issue regarding the Post structure. Basically the first Post
in a thread has a ton of extra information associated with it that the rest
of the Posts don't have (see below). Not sure what the fix for this is.
Seems like it might involve some rethinking of the Post class.

First Post Subsequent Posts
no : 12021208 no : 13125966
sub : QTDDTOT: Questions that don’t deserve their own thread resto :
12021208
com : <p clas [...] com : <p class[...]
name : Anonymous name : Anonymous
time : 1534571487 time : 1555209969
omitted_posts : 0 sticky : 0
omitted_images : 0 locked : 0
sticky : 1 cyclical : 0
locked : 0 bumplocked : 0
cyclical : 1 last_modified : 1555209969
bumplocked : 0 id : 196f83
last_modified : 1555388223
id : 674207
tn_h : 61
tn_w : 255
h : 400
w : 1680
fsize : 55627
filename : QTDDTOT
ext : .png
tim : 5fe06f6ca027c1c4[...]
fpath : 1
spoiler : 0
md5 : FY89nitoGdj+XGd2VNWg6A==
resto : 0


#12 : This seems like a good fix as well so I've included it. Basic testing
worked well.

All this should go into a pull request soon. Just wanted your opinion on
how to handle those difficulties I found in #11.
